canonical-web-and-design/jaas-dashboard
# Convert filter buttons into a ButtonGroup to toggle tables in the Model Details view...and more...

## Done

- Converted `ModelGroupToggle` component to a generic `ButtonGroup` component.
- Fixed bug in `FilterTags` that caused multiple history entries to be added if multiple components tried to update the query strings.
- Fixed bug in `LogIn` that caused the whole application to be destroyed and re-rendered on initial load.
- Created `useQueryString` hook that can be used directly by the `ButonGroup` component, or independently, to update a segment of the queryString for the component in question.
- Updated the `ModelDetails` component so that it only shows a single table depending on the selected view from the aforementioned `ButtonGroup`.
- 

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- The application should log you in as expected.
- When viewing your models list you should be able to click the grouping buttons to change the grouping and the back button should work as expected.
- After clicking a model to view its details, clicking the view buttons should change the table that's shown and the back button should work as expected.
- Viewing the model details page at different widths, the view button group should move depending on the 3 breakpoints.

## Details

Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1365

## Screenshots

![Screen Shot 2020-09-10 at 10 02 12 PM](https://user-images.githubusercontent.com/532033/92856985-4f18ae80-f3b1-11ea-82cd-2bf7ad1a2aba.png)
